src/domain/community/user/user.resolver.subscriptions.ts

```diff
@@ -30,12 +43,30 @@ export class UserResolverSubscriptions {
       value.message.sender = user?.id;
       return value;
     },
+    async filter(
+      this: UserResolverSubscriptions,
+      payload: CommunicationMessageReceived,
+      _: any,
+      context: any
+    ) {
+      // Note: by going through the passport authentication mechanism the "user" property on
+      // the request will contain the AgentInfo that was authenticated.
+      return payload.userEmail === context.req?.user?.email;
+    },
   })
-  messageReceived() {
+  async messageReceived(@CurrentUser() agentInfo: AgentInfo) {
+    const user = await this.userService.getUserOrFail(agentInfo.userID);
+    await this.authorizationEngine.grantAccessOrFail(
+      agentInfo,
+      user.authorization,
+      AuthorizationPrivilege.READ,
```

Comment:
Actually this should probably be the UPDATE privilege, which only the user has for themselves (or global admin etc)